ingest/expression_files/mtx.py

```diff
@@ -191,29 +191,34 @@ def get_data_start_line_number(file_handler: IO) -> int:
          ----------
             i (IO): Line number where data starts
         """
-        i = 0
-        for line in file_handler:
-            if line.startswith("%"):
-                i += 1
-            else:
-                # First line w/o '%' is mtx dimension. So skip this line (+1)
-                i += 2
-                return i
+        for count, line in enumerate(file_handler):
+            if not line.startswith("%"):
+                try:
+                    line_values = line.split()
+                    float(line_values[0])
```

Comment:
To help maintainability:

```suggestion
                    float(line_values[0]) # Determines if value is numeric
```